src/Http/Routing/src/RouteValuesAddress.cs

```diff
@@ -24,4 +25,10 @@ public class RouteValuesAddress
     /// Gets or sets ambient route values from the current HTTP request.
     /// </summary>
     public RouteValueDictionary? AmbientValues { get; set; }
+
+    /// <summary>
+    /// Formats the address as string "Name(ExplicitValues)" for tracing/debugging.
+    /// </summary>
+    public override string ToString() => $"{RouteName}(" + string.Join(',', ExplicitValues.ToImmutableList().ConvertAll(kv => $"{kv.Key}=[{kv.Value}]")) + ")";
```

Comment:
Is there a concrete reason where we are using `ToImmutableList` here instead of just `ExplicitValues.Select(kv => $"{kv.Key}=[{kv.Value}]")`? I don't recall using inmutable collections anywhere in the framework

Reply:
Yes, we do not want to use Linq.

Reply:
We want to use immutable collections even less. This is not a performance critical code path, so using linq should be ok.